### ISSUE TITLE
Emotion Based Stress Classification using DL

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#1004]
---
name: DL-Simplified Issue Template
about: This is the issue template that is going to be used while creating any issues
  in this repo.
title: ''
labels: ''
assignees: ''

---

### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Emotion Based Stress Classification using DL</br>
:red_circle: **Aim** : The aim of this project is to find out the stress level of a human being by their emotions using deep learning methods.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/preritbhagat/stress-non-stress-images</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by GxAditya** [comment]
@abhisheks008 can you assign me this issue if available under ssoc 2025 , Aditya Kumar | Ssoc contributor

**Comment by Mishikasardana** [comment]
@abhisheks008 can you assign me this issue if available under GSSoC's 2025, Mishika Sardana | Contributor

**Comment by deeptimaheshwari1013** [comment]
Hi, 
I'm a GSSOC'26 contributor and i would like to work on this issue. 
I'm Deepti Maheshwari. 
My Github profile link is : https://github.com/deeptimaheshwari1013
email ID: deeptimaheshwari1013@gmail.com
My approach for this project would be:
EDA
- Check for missing values
- Correlation heatmap & emotion distribution plots
- Statistical analysis (mean, std, skewness)

Preprocessing
- Label Encoding + Normalization
- Train-test split

Models
1. LSTM
2. CNN + LSTM Hybrid

Evaluation: Accuracy, F1 Score, Confusion Matrix, ROC-AUC

Stack: Python, TensorFlow/Keras, Scikit-learn, Pandas, Matplotlib

I'd like to be assigned this issue. Happy to adjust scope if needed.
ps: the kaggle dataset link isn't working.

roles: AI/Agents track

**Comment by abhisheks008** [comment]
Hi @deeptimaheshwari1013 
- Can you suggest few more models for this kind of datasets? 'cause in this repository we used to have 3-4 models for each project. 
- Also can you clarify why the models will be useful over any other models? 
- What makes these models more feasible with the dataset & why? 
- How will you check & justify the accuracy of the models?

**Comment by arijeet2006** [comment]
Full name: Arijeet Das
GitHub Profile Link: https://github.com/arijeet2006
Email ID:  2006.arijeet.das@gmail.com
Participant ID : arijeet2006
GSSOC 2026 Contrubutor
Approach for this Project:

**Exploratory Data Analysis (EDA):** Load and inspect the dataset to check the class balance between the "stress" and "non-stress" categories. Visualize sample images and analyze image dimensions to identify any anomalies.

**Data Preprocessing & Augmentation:** Resize all images to a uniform dimension (e.g., 224x224). Apply data augmentation techniques like horizontal flipping, rotation, and zooming to generalize the model and prevent overfitting. Finally, normalize the pixel values.

**Model Implementation (2 Algorithms):** I will implement and compare the following models to find the best fit:

Custom CNN: To establish a baseline performance metric.

ResNet50 (Transfer Learning): To utilize residual networks for deeper learning without vanishing gradients.

**Evaluation & Comparison**: Train the models and evaluate them using Accuracy, Precision, Recall, and F1-score.

**Comment by kashviporwal-byte** [comment]
**@abhisheks008** 

**Full name : Kashvi Porwal**

**GitHub Profile Link : https://github.com/kashviporwal-byte**

**Email ID : kashviporwal@gmail.com**

**I would like to contribute to this project under GSSoC 2026.**

**Approach for this Project :**

My planned workflow for the “Emotion Based Stress Classification using DL” project includes:

• Performing detailed Exploratory Data Analysis (EDA) on the dataset to analyze:

* Class distribution
* Image quality and dimensions
* Facial expression variations
* Dataset imbalance
* Sample visualization and annotation inspection

• Applying preprocessing and image enhancement techniques such as:

* Face detection and cropping
* Image resizing and normalization
* Noise reduction
* Contrast enhancement
* Data augmentation (rotation, flipping, zooming, brightness adjustment, etc.)

• Implementing and comparing multiple deep learning models including:

* ConvNeXt
* EfficientNetV2
* Swin Transformer
* Vision Transformer (ViT)
* DenseNet201
* Xception
* MobileNetV3

• Using transfer learning and fine-tuning techniques to improve model performance and reduce overfitting.

• Evaluating all models using important performance metrics such as:

* Accuracy
* Precision
* Recall
* F1-Score
* ROC-AUC Score
* Confusion Matrix
* Classification Report

• Creating visualizations and comparative analysis including:

* Training vs Validation Accuracy/Loss graphs
* Confusion matrices
* Model comparison charts
* Prediction visualization samples
* Feature map and Grad-CAM visualizations for interpretability

• Selecting the best-performing model based on overall accuracy, robustness, and inference efficiency.

• Organizing the repository according to contribution guidelines with:

* Images folder
* Dataset folder
* Model folder
* requirements.txt
* Properly documented README.md containing implementation details, workflow, visualizations, results, and conclusions.

• I will ensure the code is clean, modular, reproducible, and properly documented for easy understanding and future improvements.


Participant under GSSoC 2026.

**Comment by abhisheks008** [comment]
Thank you all for sharing your comments. All of them are really good approach tbh. Based on the comments of @kashviporwal-byte, this issue is assigned to you.

Happy contributing!
